File: scratch/test_upload.js
const fs = require('fs');
const path = require('path');

async function testUpload() {
  const uploadUrl = 'https://thaiba-garden-media-manager.vercel.app/api/chat/upload/';
  const mockFilePath = path.join(__dirname, 'mock_voice.txt');
  
  // Create a mock small file
  fs.writeFileSync(mockFilePath, 'This is mock voice note recorded data.');
  
  console.log('Sending mock file upload request to Vercel...');

  const FormData = require('form-data');
  const form = new FormData();
  form.append('roomId', '7bc0bbe7-1943-4929-a769-5fdfbc487446'); // General chat room ID or test ID
  form.append('file', fs.createReadStream(mockFilePath), {
    filename: 'mock_voice.m4a',
    contentType: 'audio/x-m4a'
  });

  const axios = require('axios');
  try {
    const response = await axios.post(uploadUrl, form, {
      headers: form.getHeaders(),
      timeout: 15000
    });
    console.log('Success! Status:', response.status);
    console.log('Response:', response.data);
  } catch (error) {
    console.error('Upload failed!');
    if (error.response) {
      console.error('Status:', error.response.status);
      console.error('Data:', error.response.data);
    } else {
      console.error('Error Message:', error.message);
    }
  } finally {
    // Cleanup
    try {
      fs.unlinkSync(mockFilePath);
    } catch (_) {}
  }
}

testUpload();

File: scratch/test_upload.py
import urllib.request
import json

url = "https://thaiba-garden-media-manager.vercel.app/api/chat/upload"
boundary = "----WebKitFormBoundary7MA4YWxkTrZu0gW"

# Construct multipart/form-data body
body = []
body.append(f"--{boundary}".encode('utf-8'))
body.append('Content-Disposition: form-data; name="roomId"'.encode('utf-8'))
body.append(''.encode('utf-8'))
body.append('cea4326c-213e-416c-b19f-85053fa10483'.encode('utf-8'))

body.append(f"--{boundary}".encode('utf-8'))
body.append('Content-Disposition: form-data; name="file"; filename="test_log.txt"'.encode('utf-8'))
body.append('Content-Type: text/plain'.encode('utf-8'))
body.append(''.encode('utf-8'))
body.append('This is test telemetry content.'.encode('utf-8'))
body.append(f"--{boundary}--".encode('utf-8'))
body.append(''.encode('utf-8'))

data = b'\r\n'.join(body)

headers = {
    "Content-Type": f"multipart/form-data; boundary={boundary}",
    "Content-Length": str(len(data))
}

req = urllib.request.Request(url, data=data, headers=headers, method="POST")

try:
    print("Sending upload request...")
    with urllib.request.urlopen(req) as response:
        print(f"Status Code: {response.status}")
        print("Response Body:")
        print(response.read().decode('utf-8'))
except urllib.error.HTTPError as e:
    print(f"HTTP Error: {e.code}")
    print("Response Body:")
    print(e.read().decode('utf-8'))
except Exception as e:
    print("Request failed:", e)
